src/mudlet.cpp

```diff
@@ -2879,13 +3039,80 @@ void mudlet::doAutoLogin(const QString& profile_name)
         mudlet::self()->setupPreInstallPackages(pHost->getUrl().toLower());
     }
 
+    // May be needed
+    pHost->mTelnet.setEncoding(readProfileData(profile_name, QStringLiteral("encoding")).toUtf8(), false);
+
     emit signal_hostCreated(pHost, mHostManager.getHostCount());
     emit signal_adjustAccessibleNames();
     slot_connectionDialogueFinished(profile_name, true);
     enableToolbarButtons();
     updateMultiViewControls();
 }
+void mudlet::handleTelnetUri(const QUrl& telnetUri)
+{
+    QUrl url(telnetUri);
 
+    if (url.port() == -1) {
+        url.setPort(23);
+    }
+
+    QStringList hostList = QDir(getMudletPath(profilesPath)).entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    int profilesFound{};
+    QString lastHostFound;
+    for (auto& host : hostList) {
+        QString hostUrl = readProfileData(host, QStringLiteral("url"));
+        int hostPort = readProfileData(host, QStringLiteral("port")).toInt();
+
+        if (url.host().compare(hostUrl, Qt::CaseInsensitive) == 0 && url.port() == hostPort) {
+            profilesFound++;
+            lastHostFound = host;
+        }
+    }
+
+    if (profilesFound == 0) {
+        const auto profile_name = addProfile(url.host().toLower(), url.port(), url.userName(), url.password());
+        doAutoLogin(profile_name);
+        auto pHost = getHostManager().getHost(profile_name);
+        if (!pHost) {
+            qWarning() << "no host!";
+            return;
+        }
+
+        mudlet::self()->setupPreInstallPackages(url.host().toLower());
+        mudlet::self()->installDefaultPackages(pHost);
+    } else if (profilesFound == 1) {
+        doAutoLogin(lastHostFound);
+    } else {
+        if (!mpConnectionDialog) {
+            slot_showConnectionDialog();
+        }
+        if (mpConnectionDialog) {
+            mpConnectionDialog->showInformationMessage(tr("%n matching profiles found for %1, which would you like to open?",
+                                                          "this message is shown when Mudlet is opened from a telnet:// link on a webpage, and more than one profile matches the game server/port - so "
+                                                          "the user needs to pick which of the available profiles they'd like to play with",
+                                                          profilesFound)
+                                                               .arg(url.host()));
```

Comment:
We are not using that form of providing information to the translation team (see https://github.com/Mudlet/Mudlet/issues/5522) - instead try:
```suggestion
            /*: This message is shown when Mudlet is opened from a telnet:// link on a webpage, and more than one profile matches the game server/port - so 
                the user needs to pick which of the available profiles they'd like to play with.*/
            mpConnectionDialog->showInformationMessage(tr("%n matching profiles found for %1, which would you like to open?",
                                                          "",
                                                          profilesFound)
                                                               .arg(url.host()));
```